extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/LambdaUtil.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.amazon.lambda.deployment;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import io.quarkus.deployment.pkg.builditem.OutputTargetBuildItem;
+
+public class LambdaUtil {
+
+    /**
+     * Strips period, dash, and numbers. Turns characters after to uppercase. i.e.
+     * Also strips "-SNAPSHOT" from end of name.
+     *
+     * "foo.bar-1.0-SNAPSHOT" to "FooBar"
+     *
+     * @param basename
+     * @return
+     */
+    public static String artifactToLambda(String basename) {
+        if (basename.endsWith("-SNAPSHOT"))
+            basename = basename.substring(0, basename.length() - "-SNAPSHOT".length());
+        String name = convertToken(basename, "[.0-9-]");
+        return name.trim();
+    }
```

Comment:
Should that be searching for any classifier, not just SNAPSHOT?

Reply:
Other classifiers will be pruned if they are numeric.  I just don't have enough info on the artifact, and didn't want to change quarkus core to add it.